compiler/src/main/kotlin/ast/reduce.kt

```diff
@@ -4,190 +4,190 @@ import dk.aau.cs.d409f19.antlr.CellmataParser
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.PositiveExprContext -> PositiveExpr(
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.NegativeExprContext -> NegativeExpr(
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.InverseExprContext -> InverseExpr(
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.PostIncExprContext -> PostIncExpr(
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.PostDecExprContext -> PostDecExpr(
             ctx = node,
-            value = visitExpr(node.value)
+            value = reduceExpr(node.value)
         )
         is CellmataParser.ArrayLookupExprContext -> ArrayLookupExpr(
             ctx = node,
-            index = visitExpr(node.index)
+            index = reduceExpr(node.index)
         )
         is CellmataParser.ParenExprContext -> ParenExpr(
             ctx = node,
-            expr = visitExpr(node.expr())
+            expr = reduceExpr(node.expr())
         ) // ToDo: Should we flatten this?
-        is CellmataParser.LiteralExprContext -> visitExpr(node.value)
+        is CellmataParser.LiteralExprContext -> reduceExpr(node.value)
         is CellmataParser.VarExprContext -> NamedExpr(
             ctx = node,
             ident = node.ident.text
         )
         is CellmataParser.FuncExprContext -> FuncExpr(
             ctx = node,
-            args = node.value.expr().map { visitExpr(it) }
+            args = node.value.expr().map(::reduceExpr)
         )
         is CellmataParser.StateIndexExprContext -> StateIndexExpr(ctx = node)
         is CellmataParser.ArrayValueExprContext -> ArrayBodyExpr(
             ctx = node,
-            values = node.array_value().array_body().expr().map(::visitExpr)
+            values = node.array_value().array_body().expr().map(::reduceExpr)
         )
-        is CellmataParser.NumberLiteralContext -> visitExpr(node.value)
-        is CellmataParser.BoolLiteralContext -> visitExpr(node.value)
+        is CellmataParser.NumberLiteralContext -> reduceExpr(node.value)
+        is CellmataParser.BoolLiteralContext -> reduceExpr(node.value)
         is CellmataParser.Bool_literalContext -> BoolLiteral(ctx = node)
-        is CellmataParser.Number_literalContext -> visitExpr(node.getChild(0))
-        is CellmataParser.IntegerLiteralContext -> visitExpr(node.value)
-        is CellmataParser.FloatLiteralContext -> visitExpr(node.value)
+        is CellmataParser.Number_literalContext -> reduceExpr(node.getChild(0)) // ToDo is this correct?
```

Comment:
```suggestion
        is CellmataParser.Number_literalContext -> reduceExpr(node.getChild(0))
```

Reply:
I previously removed this comment because i determined `node.getChilde(0)` to be correct